go/pserver/service.go

```diff
@@ -186,13 +231,13 @@ func (s *Service) doCheckpoint() error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	cp := make([]ParameterCheckpoint, 0, len(s.optMap))
+	cp := make([]parameterCheckpoint, 0, len(s.optMap))
```

Comment:
This does not work. It creates a slice with length set to 0 and capacity set to `len(s.optMap)`. access `cp[0]` will crash the program since it's length is 0. Should use `cp := make([]parameterCheckpoint, len(s.optMap))` here.

For example:

```go
package main

func main() {
	a := make([]int, 0, 10)
	for i := 0; i < 10; i++ {
		a[i] = i
	}
}
```
Output:
```text
panic: runtime error: index out of range

goroutine 1 [running]:
main.main()
	/tmp/sandbox064034874/main.go:6 +0x20
```